### PR TITLE
 Add new components & Modified lobby, sign-up component

### PR DESCRIPTION
**Pull request by edwardycl**
state쪽은 거의 다 sign-in에 해당하는거 copy한거니까 확인필수

### CODE REVIEW

File: frontend/src/app/lobby.component.ts [hunk 1]
```diff
@@ -24,4 +29,23 @@ export class LobbyComponent implements OnInit {
   ngOnInit() {
   }
 
+  gotoCreateGame() {
+    if (!this.signedIn) {
```
Comment (tirr-c) [inline frontend/src/app/lobby.component.ts]: `UserActions.RedirectWithSignInState`가 있음! [여기]에 사용 예시가 있으니 참고. `signed-in` 대신 `not-signed-in`을 넣으면 로그인되지 않았을 때 리디렉션할 수 있음.

[여기]: https://github.com/swsnu/swpp17-team2/blob/master/frontend/src/app/sign-in.component.ts#L35

File: frontend/src/app/menu-bar.component.ts [hunk 2]
```diff
@@ -32,7 +33,9 @@ export class MenuBarComponent implements OnInit {
     this.store.dispatch(new RouterActions.GoByUrl('lobby'));
   }
   gotoProfile() {
-    // this.router.navigateByUrl('profile/someone');
+    let usernameString: string;
+    this.username.subscribe(username => usernameString = username);
+    this.store.dispatch(new RouterActions.Go({path: ['profile', usernameString]}));
```
Comment (tirr-c) [inline frontend/src/app/menu-bar.component.ts]:
```typescript
this.username.first().subscribe(username =>
  this.store.dispatch(new RouterActions.Go({ path: ['profile', username] })
);
```

File: frontend/src/app/sign-up.component.html [hunk 3]
```diff
@@ -21,5 +21,6 @@
     <input [(ngModel)]="nickname" name="nickname" type="text" [placeholder]="username">
   </div>
   <button (click)="signUp()">Sign up</button>
+  <div *ngIf="retry">Passwords don't match. Try again.</div>
```
Comment (tirr-c) [inline frontend/src/app/sign-up.component.html]: `UserActions.SignUp.Failed`로 합칠 수 없을까?

File: frontend/src/app/state/effects/user.ts [hunk 4]
```diff
@@ -31,11 +34,34 @@ export class UserEffects {
...
     .map((action: UserActions.SignUp.Start) => action.payload)
-    .mergeMap(({ email, username, password }) =>
-      Observable.of(new UserActions.SignUp.Done())
+    .mergeMap(params =>
+      this.http.post(
+        '/api/signup/',
+        JSON.stringify(params),
+        { headers: UserEffects.jsonHeaders }
+      ).mergeMap((response): Observable<Action> => {
+        if(response.status !== 201) {
```
Comment (tirr-c) [inline frontend/src/app/state/effects/user.ts]: `!response.ok`로 쓰자. 이거 고친 거 아직 머지 안 됐음...

**Comment by web-mighty-homu** [comment]
:umbrella: The latest upstream changes (presumably #27) made this pull request unmergeable. Please resolve the merge conflicts.

**Comment by tirr-c** [comment]
@web-mighty-homu r+

**Comment by tirr-c** [comment]
@web-mighty-homu r+

**Comment by web-mighty-homu** [comment]
:pushpin: Commit 5a6eab8 has been approved by `tirr-c`

**Comment by web-mighty-homu** [comment]
:pushpin: Commit 663f205 has been approved by `tirr-c`

**Comment by tirr-c** [comment]
@web-mighty-homu r+

**Comment by web-mighty-homu** [comment]
:bulb: This pull request was already approved, no need to approve it again.

**Comment by web-mighty-homu** [comment]
:hourglass: Testing commit 663f20509fe9c592e64aa99c812bfe339ddd144b with merge 71a12024179a9ceb3ef4c09b1df0714f2c49ae93...

**Comment by web-mighty-homu** [comment]
:pushpin: Commit 663f205 has been approved by `tirr-c`

**Comment by web-mighty-homu** [comment]
:hourglass: Testing commit 663f20509fe9c592e64aa99c812bfe339ddd144b with merge 8a50cd1647b91a4b7ef3d8118149710346d896af...

**Comment by web-mighty-homu** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/swsnu/swpp17-team2/builds/302415612?utm_source=github_status&utm_medium=notification)
Approved by: tirr-c
Pushing 8a50cd1647b91a4b7ef3d8118149710346d896af to master...

**Pull request merged**